src/main/java/lesson44/dao/StudentDao.java

```diff
@@ -0,0 +1,37 @@
+package lesson44.dao;
+
+import lesson44.model.StudentModel;
+import lesson44.repository.StudentsRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Component
+public class StudentDao {
+    @Autowired
+    private StudentsRepository repository;
```

Comment:
Repository and DAO are different names for the same layer if the application.
They cannot inject each other.
Rename one of them